### ISSUE TITLE
Default AZCOPY_JOB_PLAN_LOCATION leads to panic: mkdir /home/app/.azcopy: read-only file system

### ISSUE THREAD

**Issue by sandersaares** [Azure/azure-storage-azcopy#2191]
### Which version of the AzCopy was used? 
Whatever downloading from [the v10 URL](https://aka.ms/downloadazcopy-v10-linux) installed today. AzCopy fails to run even without any argument - cannot check for sure.

### Which platform are you using? (ex: Windows, Mac, Linux)
Linux

### What command did you run?
1. `azcopy`
2. Receive error saying "You should point AZCOPY_LOG_LOCATION to writable directory`
3. Set `AZCOPY_LOG_LOCATION=/tmp/azcopy`
4. `azcopy`
5. Receive new error without helpful instructions to resolve.

### What problem was encountered?
App immediately crashes with:

```
panic: mkdir /home/app/.azcopy: read-only file system

goroutine 1 [running]:
github.com/Azure/azure-storage-azcopy/v10/common.PanicIfErr(...)
        /home/vsts/work/1/s/common/lifecyleMgr.go:696
main.main()
        /home/vsts/work/1/s/main.go:63 +0x57a
```

### How can we reproduce the problem in the simplest way?
1. Create a container with azcopy, mount with a read-only filesystem and writable /tmp
2. Point AZCOPY_LOG_LOCATION to writable location
3. Yry to execute `azcopy`.

### Have you found a mitigation/solution?
It seems I also need to define `AZCOPY_JOB_PLAN_LOCATION` and point to a writable location.

There is an error message in the code for `AZCOPY_JOB_PLAN_LOCATION` but it is never reached for some reason.

https://github.com/Azure/azure-storage-azcopy/blob/053a04973c31c137fc86bd3c55dac5417cdac96f/main.go#L69

When I set `AZCOPY_JOB_PLAN_LOCATION` to a writable location, the error goes away. I suggest investigating why the app crashes before the error message about `AZCOPY_JOB_PLAN_LOCATION` is displayed, and ensuring the error actually gets displayed.

**Comment by gapra-msft** [comment]
Hi @sandersaares thank you for reporting this issue, we will work on reproducing this behavior and potentially wrap the error in a more helpful message.

**Comment by nakulkar-msft** [comment]
Fixed in 10.19.0
